ci/unit_tests.sh

```diff
@@ -19,11 +19,19 @@ SELECTED_TEST_SUITE=$1
 
 if [[ $SELECTED_TEST_SUITE == $"java" ]]; then
   echo "Running Java Tests"
-  ./gradlew javaTests --console=plain --warning-mode all
+  ./gradlew javaTests jacocoTestReport sonar -Dsonar.token="${SONAR_TOKEN}" \
+    -Dsonar.host.url=https://sonar.elastic.dev \
+    -Dsonar.projectKey=elastic_logstash_AYm_nEbQaV3I-igkX1q9 \
+    -Dsonar.projectName=logstash \
+    -Dsonar.pullrequest.key=$PULL_ID \
+    -Dsonar.pullrequest.branch=$SOURCE_BRANCH \
+    -Dsonar.pullrequest.base=$TARGET_BRANCH \
+    -Dsonar.scm.revision=$COMMIT_SHA \
+    --console=plain --warning-mode all
 elif [[ $SELECTED_TEST_SUITE == $"ruby" ]]; then
   echo "Running Ruby unit tests"
   ./gradlew rubyTests --console=plain --warning-mode all
 else
   echo "Running Java and Ruby unit tests"
-  ./gradlew test --console=plain
+  ./gradlew test jacocoTestReport sonar -Dsonar.token="${SONAR_TOKEN}" --console=plain --warning-mode all
```

Comment:
Our complex build system strikes again... 😩 

Currently this [fails](https://logstash-ci.elastic.co/job/elastic+logstash+main+multijob-unix-compatibility/os=amazon/412/) the Unix acceptance tests.

The Unix acceptance tests call `ci/unit_tests.sh` directly without a parameter, and also bypass the `ci/docker_unit_tests.sh` script.

This gives us a couple of problems - firstly, the system properties are not being set in this code path, so the sonar server cannot be found. Secondly, as we are bypassing the docker script, we're not setting the `ENV` variables.

I guess we could just call the `sonar` and `jacocoTestReport` tasks when we explicitly specify the language type... Or add a command line option to include/exclude running those tests, and update the jenkins jobs accordingly

Reply:
Updated the pr. The workaround should work now